modules/programs/mujmap.nix

```diff
@@ -0,0 +1,312 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mujmap;
+
+  mujmapAccounts =
+    filter (a: a.mujmap.enable) (attrValues config.accounts.email.accounts);
+
+  missingNotmuchAccounts = map (a: a.name)
+    (filter (a: !a.notmuch.enable && a.mujmap.notmuchSetupWarning)
+      mujmapAccounts);
+
+  notmuchConfigHelp =
+    map (name: "accounts.email.accounts.${name}.notmuch.enable = true;")
+    missingNotmuchAccounts;
+
+  fqdnForFlavor = flavor:
+    if flavor == "fastmail.com" then "fastmail.com" else null;
```

Comment:
It may be better to add support for jmap in `accounts/email.nix` and then add an entry for jmap in the

```nix
mkIf (config.flavor == "fastmail.com") { … }
```

block. Then that configuration could be used in other modules that support JMAP.

Reply:
that's a good idea, the same way we have an imapModule/smtpModule, we could have a jmapModule.